### PR TITLE
mds: misc snap fixes

### PR DESCRIPTION
**Pull request by ukernel**
misc fixes and cleanup in https://github.com/ceph/ceph/pull/16779

### CODE REVIEW

**Comment by ukernel** [comment]
http://pulpito.ceph.com/zyan-2017-11-23_00:48:00-fs-wip-zyan-integration-20171122-testing-basic-mira/

File: src/mds/CInode.cc [hunk 1]
```diff
@@ -1400,6 +1419,8 @@ void CInode::set_object_info(MDSCacheObjectInfo &info)
 void CInode::encode_lock_state(int type, bufferlist& bl)
 {
   ::encode(first, bl);
+  if (!is_base())
+    ::encode(parent->first, bl);
```
Comment (batrick) [inline src/mds/CInode.cc]: This requires bumping `CEPH_MDS_PROTOCOL`?

Comment (ukernel) [reply src/mds/CInode.cc]: yes,

**Comment by ukernel** [comment]
updated

**Comment by ukernel** [comment]
updated

**Pull request merged**